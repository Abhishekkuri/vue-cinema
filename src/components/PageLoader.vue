<template>
  <div class="loader hypnotic"></div>
</template>

<script>
export default {};
</script>

<style lang="scss">
.hypnotic {
  width: 56px;
  height: 56px;
  border: 2.2px solid $color-primary;
  display: grid;
  animation: hypnotic-q3fddx 4s infinite linear;
}

.hypnotic::before,
.hypnotic::after {
  content: "";
  grid-area: 1/1;
  margin: auto;
  width: 70.7%;
  height: 70.7%;
  border: 2.2px solid $color-primary;
  animation: inherit;
}

.hypnotic::after {
  width: 50%;
  height: 50%;
  border: 2.2px solid $color-primary;
  animation-duration: 2s;
}

@keyframes hypnotic-q3fddx {
  100% {
    transform: rotate(1turn);
  }
}
</style>
