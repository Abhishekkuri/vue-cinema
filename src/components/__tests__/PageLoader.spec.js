import { shallowMount } from "@vue/test-utils";
import PageLoader from "@/components/PageLoader";

describe("PageLoader.vue", () => {
  test("loader renders correctly", () => {
    expect(shallowMount(PageLoader).element).toMatchSnapshot();
  });
});
