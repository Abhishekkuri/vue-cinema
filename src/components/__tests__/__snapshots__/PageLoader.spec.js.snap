// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageLoader.vue loader renders correctly 1`] = `
<div
  class="loader hypnotic"
/>
`;
